src/common/StackStringStream.h

```diff
@@ -77,6 +77,8 @@ class StackStringBuf : public std::basic_streambuf<char>
     if (traits_type::not_eof(c)) {
       char str = traits_type::to_char_type(c);
       vec.push_back(str);
+      setp(vec.data(), vec.data() + vec.size());
```

Comment:
this mirrors the overflow handling in `xsputn()`, which is sane. so it looks good as well.